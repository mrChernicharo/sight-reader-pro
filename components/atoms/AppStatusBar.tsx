import { useTheme } from "@/hooks/useTheme";
import { StatusBar } from "expo-status-bar";

export function AppStatusBar() {
    const theme = useTheme();
    const style = theme == "light" ? "dark" : "light";

    return <StatusBar translucent style={style} />;
}
