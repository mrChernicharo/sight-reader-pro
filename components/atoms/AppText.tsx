import { Text, type TextProps, StyleSheet } from "react-native";

import { useThemeColor } from "@/hooks/useThemeColor";
import { Colors } from "@/utils/Colors";

type TextType = "default" | "defaultSemiBold" | "md" | "mdSemiBold" | "lg" | "title" | "subtitle" | "link";

export type AppTextProps = TextProps & {
    lightColor?: string;
    darkColor?: string;
    type?: TextType;
    forceBlackText?: boolean;
};

export function AppText({ style, lightColor, darkColor, forceBlackText, type, ...rest }: AppTextProps) {
    // const color = useThemeColor({ light: lightColor, dark: darkColor }, "text");
    const themeColor = useThemeColor(
        { light: (lightColor = Colors.light.text), dark: (darkColor = Colors.dark.text) },
        "text"
    );
    const color = forceBlackText ? "black" : themeColor;

    const textStyle = s[type!] ? s[type!] : s.default;

    return <Text style={[{ color }, textStyle, style]} {...rest} />;
}

const s = StyleSheet.create({
    default: {
        fontSize: 14,
        lineHeight: 26,
    },
    defaultSemiBold: {
        fontSize: 14,
        lineHeight: 26,
        fontWeight: "600",
    },
    md: {
        fontSize: 16,
        lineHeight: 24,
    },
    mdSemiBold: {
        fontSize: 16,
        lineHeight: 24,
        fontWeight: "600",
    },
    lg: {
        fontSize: 24,
        lineHeight: 32,
    },
    title: {
        fontSize: 28,
        lineHeight: 36,
        fontWeight: "bold",
    },
    subtitle: {
        fontSize: 20,
        lineHeight: 36,
        fontWeight: "bold",
    },
    link: {
        lineHeight: 30,
        fontSize: 16,
        color: "#0a7ea4",
    },
});
