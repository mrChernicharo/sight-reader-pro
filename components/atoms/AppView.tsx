import { View, type ViewProps } from "react-native";

import { useThemeColor } from "@/hooks/useThemeColor";
import { Colors } from "@/utils/Colors";

export type AppViewProps = ViewProps & {
    lightColor?: string;
    darkColor?: string;
    transparentBG?: boolean;
};

export function AppView({ style, lightColor, darkColor, transparentBG, ...otherProps }: AppViewProps) {
    const themeBG = useThemeColor({ light: (lightColor = Colors.light.bg), dark: (darkColor = Colors.dark.bg) }, "bg");

    const backgroundColor = transparentBG ? "transparent" : themeBG;

    return <View style={{ backgroundColor, ...(style as any) }} {...otherProps} />;
}
