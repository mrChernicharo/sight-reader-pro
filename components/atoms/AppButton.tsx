import React, { forwardRef, ReactNode } from "react";
import { TextStyle, TouchableOpacity, TouchableOpacityProps, View, ViewStyle } from "react-native";
import { AppText } from "./AppText";

interface AppButtonProps extends TouchableOpacityProps {
    text: string;
    icon?: ReactNode;
    style?: ViewStyle;
    textStyle?: TextStyle;
    disabled?: boolean;
}

const AppButton = forwardRef<View, AppButtonProps>((props, ref) => {
    const { text, icon, style, textStyle, disabled, ...otherProps } = props;

    const disabledStyles = { backgroundColor: disabled ? "gray" : "blue", opacity: disabled ? 0.5 : 1 };
    const buttonStyle = [defaultStyles, disabledStyles, style] as ViewStyle[];

    const Icon = () => (icon ? icon : null);

    return (
        <TouchableOpacity ref={ref} style={buttonStyle} disabled={disabled} {...otherProps}>
            <Icon />
            <AppText style={{ color: "white", ...textStyle }}>{text}</AppText>
        </TouchableOpacity>
    );
});

const defaultStyles = {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    gap: 6,
    borderRadius: 5,
    paddingHorizontal: 16,
    paddingVertical: 8,
};

export default AppButton;
