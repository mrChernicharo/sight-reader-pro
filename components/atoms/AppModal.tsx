import { ReactNode } from "react";
import { Pressable } from "react-native";
import Modal from "react-native-modal";
import { AppText } from "./AppText";
import { AppView } from "./AppView";

interface IModalProps {
    isOpen: boolean;
    setOpen: (open: boolean) => void;
    children: ReactNode;
}

export function AppModal({ isOpen, setOpen, children }: IModalProps) {
    const toggleModal = () => {
        setOpen(!isOpen);
    };

    return (
        <AppView style={{ flex: 1 }}>
            <Modal
                isVisible={isOpen}
                // deviceWidth={Dimensions.get("window").width}
                // deviceHeight={Dimensions.get("window").height}
            >
                <AppView style={{ flex: 1, borderRadius: 16, padding: 16 }}>
                    <Pressable onPress={toggleModal}>
                        <AppText>x</AppText>
                    </Pressable>
                    {/* <Button title="Hide modal" onPress={toggleModal} /> */}
                    {children}
                </AppView>
            </Modal>
        </AppView>
    );
}
