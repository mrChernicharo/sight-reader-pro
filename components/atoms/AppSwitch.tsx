import { StyleSheet, Switch } from "react-native";
import { AppView } from "./AppView";

export function AppSwitch({ value, setValue }: { value: boolean; setValue: (val: boolean) => void }) {
    return (
        <Switch
            trackColor={{ false: "#767577", true: "#767577" }}
            thumbColor="#81b0ff"
            ios_backgroundColor="#3e3e3e"
            onValueChange={(val) => setValue(val)}
            value={value}
        />
    );
}

const s = StyleSheet.create({});
