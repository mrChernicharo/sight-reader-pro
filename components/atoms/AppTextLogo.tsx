import { testBorder } from "@/utils/styles";
import { AppLogo } from "./AppLogo";
import { AppText } from "./AppText";
import { AppView } from "./AppView";
import { APP_NAME } from "@/utils/constants";

export function AppTextLogo({ subtitles }: { subtitles?: string }) {
    return (
        <AppView transparentBG style={{ paddingRight: 20 }}>
            <AppView transparentBG style={{ flexDirection: "row", alignItems: "center", marginBottom: 12 }}>
                <AppLogo />
                <AppText type="title" style={{ marginLeft: -2, marginTop: 4 }}>
                    {APP_NAME}
                </AppText>
            </AppView>

            {subtitles ? (
                <AppView style={{ width: "110%", marginTop: -16 }}>
                    <AppText style={{ textAlign: "center" }}>{subtitles}</AppText>
                </AppView>
            ) : null}
        </AppView>
    );
}
