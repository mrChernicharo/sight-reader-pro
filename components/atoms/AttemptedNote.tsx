import { AppText } from "./AppText";
import { AttemptedNote as AttemptedNoteType } from "@/utils/types";
import { FadeOut } from "./FadeOut";
import { explodeNote, getAttemptedNoteDuration, getNoteIdx, isNoteMatch, mapRange } from "@/utils/helperFns";
import { useTranslation } from "@/hooks/useTranslation";
import { useThemeColor } from "@/hooks/useThemeColor";
import { Colors } from "@/utils/Colors";
import { useTheme } from "@/hooks/useTheme";
import { testBorder } from "@/utils/styles";
import AntDesign from "@expo/vector-icons/build/AntDesign";
import { AppView } from "./AppView";
import { NoteName } from "@/utils/enums";
import { Dimensions } from "react-native";

const calcOffset = (zeroToOneValue: number) => {
    return mapRange(zeroToOneValue, 0, Dimensions.get("window").width, 1);
};

const noteStyles: Record<NoteName, any> = {
    c: { left: calcOffset(0.0125) },
    "b#": { left: calcOffset(0.0125) },
    dbb: { left: calcOffset(0.0125) },

    "c#": { left: calcOffset(0.065) },
    db: { left: calcOffset(0.065) },

    d: { left: calcOffset(0.125) },
    cx: { left: calcOffset(0.125) },
    ebb: { left: calcOffset(0.125) },

    "d#": { left: calcOffset(0.2) },
    eb: { left: calcOffset(0.2) },

    e: { left: calcOffset(0.265) },
    dx: { left: calcOffset(0.265) },
    fb: { left: calcOffset(0.265) },

    f: { right: calcOffset(0.4) },
    "e#": { right: calcOffset(0.4) },

    gbb: { right: calcOffset(0.325) },
    "f#": { right: calcOffset(0.325) },
    gb: { right: calcOffset(0.325) },

    g: { right: calcOffset(0.265) },
    fx: { right: calcOffset(0.265) },
    abb: { right: calcOffset(0.265) },

    "g#": { right: calcOffset(0.2) },
    ab: { right: calcOffset(0.2) },

    a: { right: calcOffset(0.125) },
    gx: { right: calcOffset(0.125) },
    bbb: { right: calcOffset(0.125) },

    "a#": { right: calcOffset(0.065) },
    bb: { right: calcOffset(0.065) },

    b: { right: calcOffset(0.0125) },
    ax: { right: calcOffset(0.0125) },
    cb: { right: calcOffset(0.0125) },
};

export function AttemptedNote({ attempt }: { attempt: AttemptedNoteType }) {
    const { t } = useTranslation();
    const theme = useTheme();

    const { you, correct, isSuccess, noteScore } = attempt;

    const { noteName } = explodeNote(you);
    // const { noteName: correct } = explodeNote(attempt.correct);

    const duration = getAttemptedNoteDuration(isSuccess);
    const color = isSuccess ? Colors[theme].green : Colors[theme].red;
    const notePositionStyles = noteStyles[noteName];

    return (
        <FadeOut
            y={-60}
            duration={duration}
            style={{
                position: "absolute",
                alignItems: "center",
                bottom: 12,
                ...notePositionStyles,
            }}
        >
            {/* TOP LINE */}
            <AppView transparentBG style={{ flexDirection: "row", alignItems: "center", gap: 2 }}>
                <AntDesign
                    name={isSuccess ? "checkcircle" : "closecircle"}
                    style={{ fontWeight: 900, marginBottom: 2 }}
                    size={14}
                    color={color}
                />

                <AppText style={{ fontFamily: "Grotesque", fontSize: 24, lineHeight: 36 }}>
                    {t(`music.notes.${noteName}`)}
                </AppText>
            </AppView>

            {/* BOTTOM LINE */}
            <AppText style={{ fontFamily: "Grotesque", lineHeight: 18 }}>
                {isSuccess
                    ? `${noteScore} pts`
                    : `( ${t(`game.was`)} ${t(`music.notes.${explodeNote(correct).noteName}`)} )`}
            </AppText>
        </FadeOut>
    );
}
