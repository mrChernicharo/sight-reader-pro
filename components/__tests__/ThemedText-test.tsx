import * as React from "react";
import renderer from "react-test-renderer";

import { AppText } from "../atoms/AppText";

it(`renders correctly`, () => {
    const tree = renderer.create(<AppText>Snapshot test!</AppText>).toJSON();

    expect(tree).toMatchSnapshot();
});
