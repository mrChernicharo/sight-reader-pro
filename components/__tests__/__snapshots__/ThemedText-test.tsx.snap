// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#151718",
      },
      {
        "fontSize": 14,
        "lineHeight": 26,
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
